src/GoogleFit/FitOnFhir.GoogleFit.Tests/GoogleFitImportServiceTests.cs

```diff
@@ -0,0 +1,221 @@
+        private const string _deviceUid = "DeviceUid";
+        private const string _applicationPackageName = "ApplicationPackageName";
+
+        private readonly AuthTokensResponse _tokensResponse = new AuthTokensResponse() { AccessToken = _accessToken, RefreshToken = _refreshToken };
+        private readonly CancellationToken _cancellationToken = CancellationToken.None;
+
+        private readonly Dataset _dataset = new Dataset();
+        private readonly DataSource _dataSource = new DataSource(_dataStreamId, _deviceUid, _applicationPackageName);
+        private readonly List<DataSource> _dataSources = new List<DataSource>();
+
+        private readonly DateTimeOffset _now;
+        private readonly DateTimeOffset _oneDayBack = DateTimeOffset.Parse("01/11/2004");
+
+        private readonly GoogleFitUser _googleFitUser;
+        private readonly MockEventHubProducerClient _eventHubProducerClient;
+        private readonly MockFaultyEventHubProducerClient _faultyEventHubProducerClient;
+        private readonly EventData _eventData;
+        private readonly MedTechDataset _medTechDataset;
+
+        private readonly IGoogleFitClient _googleFitClient;
+        private readonly GoogleFitImportOptions _options;
+        private readonly Func<DateTimeOffset> _utcNowFunc;
+        private readonly MockLogger<GoogleFitImportService> _importServiceLogger;
+        private readonly ITelemetryLogger _telemetryLogger;
+        private IGoogleFitImportService _googleFitImportService;
+
+        public GoogleFitImportServiceTests()
+        {
+            // _googleFitUser = new GoogleFitUser(_googleUserId);
+            _googleFitUser = Substitute.For<GoogleFitUser>(_googleUserId);
+            _eventData = new EventData(_eventBody);
+            _medTechDataset = new MedTechDataset(_dataset, _dataSource);
+            _dataSources.Add(_dataSource);
+            _faultyEventHubProducerClient = new MockFaultyEventHubProducerClient();
+
+            // GoogleFitImportService dependencies
+            _googleFitClient = Substitute.For<IGoogleFitClient>();
+            _eventHubProducerClient = new MockEventHubProducerClient();
+            _options = Substitute.For<GoogleFitImportOptions>();
+            _utcNowFunc = Substitute.For<Func<DateTimeOffset>>();
+            _now = DateTimeOffset.Parse("01/12/2004");
+            _utcNowFunc().Returns(_now);
+            _importServiceLogger = Substitute.For<MockLogger<GoogleFitImportService>>();
+            _telemetryLogger = Substitute.For<ITelemetryLogger>();
+
+            // create the service
+            _googleFitImportService = new GoogleFitImportService(
+                _googleFitClient,
+                _eventHubProducerClient,
+                _options,
+                _utcNowFunc,
+                _importServiceLogger,
+                _telemetryLogger);
+        }
+
+        [Fact]
+        public async Task GivenNoLastSyncStored_WhenTryGetLastSyncTime_GeneratesDataSetIdForLast30Days()
```

Comment:
This test fails also:

`NSubstitute.Exceptions.ReceivedCallsException : Expected to receive exactly 1 call matching:
	DatasetRequest(access => (access == value(FitOnFhir.GoogleFit.Tests.GoogleFitImportServiceTests)._tokensResponse.AccessToken), ds => (ds == value(FitOnFhir.GoogleFit.Tests.GoogleFitImportServiceTests)._dataSource), str => (str == value(FitOnFhir.GoogleFit.Tests.GoogleFitImportServiceTests+<>c__DisplayClass26_0).thirtyDaysBackDataSetId), token => (token == value(FitOnFhir.GoogleFit.Tests.GoogleFitImportServiceTests)._cancellationToken), <null>)
Actually received no matching calls.
Received 1 non-matching call (non-matching arguments indicated with '*' characters):
	DatasetRequest("AccessToken", DataSource, *"1071302400000000000-1073894400000000000"*, System.Threading.CancellationToken, <null>)`